src/store/RelayQueryWriter.js

```diff
@@ -552,7 +552,7 @@ class RelayQueryWriter extends RelayQueryVisitor<WriterState> {
     this._store.putLinkedRecordIDs(recordID, storageKey, nextLinkedIDs);
```

Comment:
Can you move the logic here so that it's more clear what is stored into the `isUpdated` variable and we only need to check one thing?
`isUpdate = isUpdate ||  prevLinkedIDs.length !== nextLinkedIDs.length`